CheckDrive.Api/CheckDrive.Application/Services/Authorization/AuthService.cs

```diff
@@ -0,0 +1,106 @@
+﻿using AutoMapper;
+using CheckDrive.Application.Constants;
+using CheckDrive.Application.DTOs.Identity;
+using CheckDrive.Application.Interfaces.Authorization;
+using CheckDrive.Domain.Exceptions;
+using Microsoft.AspNetCore.Identity;
+
+namespace CheckDrive.Application.Services.Authorization;
+
+public class AuthService : IAuthService
+{
+    private readonly JwtHandler _jwtHandler;
+
+    private readonly UserManager<IdentityUser> _userManager;
+
+    private readonly RoleManager<IdentityRole> _roleManager;
+
+    private readonly IMapper _mapper;
+
+    private readonly IEmployeeRegistrationService _employeeFactory;
+
+    public AuthService(
+        IMapper mapper,
+        JwtHandler jwtHandler,
+        UserManager<IdentityUser> userManager,
+        RoleManager<IdentityRole> roleManager,
+        IEmployeeRegistrationService employeeFactory
+    )
+    {
+        _jwtHandler = jwtHandler 
+            ?? throw new ArgumentNullException(nameof(jwtHandler));
+        _userManager = userManager 
+            ?? throw new ArgumentNullException(nameof(userManager));
+        _roleManager = roleManager 
+            ?? throw new ArgumentNullException(nameof(roleManager));
+        _mapper = mapper 
+            ?? throw new ArgumentNullException(nameof(mapper));
+        _employeeFactory = employeeFactory 
+            ?? throw new ArgumentNullException(nameof(employeeFactory));
+    }
+
+    public async Task<string> LoginAsync(LoginDto loginDto)
+    {
+        var user = await _userManager.FindByEmailAsync(loginDto.Email)
+            ?? throw new InvalidLoginAttemptException("Invalid email or password");
+
+        ValidateUserForLogin(user, loginDto.Password);
+
+        return await _jwtHandler.GenerateTokenAsync(user);
```

Comment:
we should not use custom token handler for token generation, userManager already provides this functionality for us.